src/index.js

```diff
@@ -235,7 +235,7 @@ export class Helmet extends Component {
 
     return helmetData ? (
       // eslint-disable-next-line react/jsx-props-no-spreading
-      <Dispatcher {...newProps} context={helmetData.value} />
+      <Dispatcher {...newProps} context={helmetData.value} helmetData={undefined} />
```

Comment:
Don't propagate the `helmetData` do the Dispatcher, this makes it so that the dispatcher doesn't also trigger changes in `shouldComponentUpdate`